Aztec/Classes/Converters/NSAttributedStringToNodes.swift

```diff
@@ -487,83 +535,133 @@ private extension NSAttributedStringToNodes {
+    private func processLinkStyle(in attributes: [String: Any]) -> ElementNode? {
         guard let url = attributes[NSLinkAttributeName] as? URL else {
-            return []
+            return nil
         }
 
-        let representation = attributes[LinkFormatter.htmlRepresentationKey] as? HTMLElementRepresentation
-        let node = representation?.toNode() ?? ElementNode(type: .a)
-        node.updateAttribute(named: "href", value: url.absoluteString)
+        let element: ElementNode
+
+        if let representation = attributes[LinkFormatter.htmlRepresentationKey] as? HTMLRepresentation,
+            case let .element(representationElement) = representation {
+
+            element = representationElement.toElementNode()
+        } else {
+            element = ElementNode(type: .a)
+        }
 
-        return [node]
+        element.updateAttribute(named: "href", value: .string(url.absoluteString))
+
+        return element
     }
 
 
     /// Extracts all of the Strike Elements contained within a collection of Attributes.
     ///
-    private func processStrikethruStyle(in attributes: [String: Any]) -> [ElementNode] {
+    private func processStrikethruStyle(in attributes: [String: Any]) -> ElementNode? {
         guard attributes[NSStrikethroughStyleAttributeName] != nil else {
-            return []
+            return nil
         }
 
-        let representation = attributes[StrikethroughFormatter.htmlRepresentationKey] as? HTMLElementRepresentation
-        let node = representation?.toNode() ?? ElementNode(type: .strike)
+        let element: ElementNode
+
+        if let representation = attributes[StrikethroughFormatter.htmlRepresentationKey] as? HTMLRepresentation,
+            case let .element(representationElement) = representation {
+
+            element = representationElement.toElementNode()
+        } else {
+            element = ElementNode(type: .strike)
+        }
 
-        return [node]
+        return element
     }
 
 
     /// Extracts all of the Underline Elements contained within a collection of Attributes.
     ///
-    private func processUnderlineStyle(in attributes: [String: Any]) -> [ElementNode] {
+    private func processUnderlineStyle(in attributes: [String: Any]) -> ElementNode? {
         guard attributes[NSUnderlineStyleAttributeName] != nil else {
-            return []
+            return nil
         }
 
-        let representation = attributes[UnderlineFormatter.htmlRepresentationKey] as? HTMLElementRepresentation
-        let node = representation?.toNode() ?? ElementNode(type: .u)
+        let element: ElementNode
+
+        if let representation = attributes[UnderlineFormatter.htmlRepresentationKey] as? HTMLRepresentation,
+            case let .element(representationElement) = representation {
+
+            element = representationElement.toElementNode()
+        } else {
+            element = ElementNode(type: .u)
+        }
```

Comment:
Remember Diego: the goal of this PR is to have an equal number of (-) and (+) changes. Please consider my suggestion, since you're just +7 away from the target!